### PR TITLE
Added API route to change a user's password #245

### LINKED ISSUE #245: As a developer, I want to have access to a Change Password endpoint, so that can allow users to securely update their passwords.
## Acceptance Criteria:  
- [ ] Endpoint for password change is functional and secure.
- [ ] User identity is verified before allowing password change.
- [ ] Endpoint integrates smoothly with both Forgot Password and Settings pages.
- [ ] Process adheres to security best practices, including token generation.

## Demo Steps:  
1. Show password change via Forgot Password page.
2. Demonstrate password change in Settings for logged-in users.
3. Confirm user verification process.

## Potential Risks:  
- Security vulnerabilities in the password change process.
- User confusion if the process is not intuitive.
- Integration challenges with existing systems.

## Technical Notes:  
- Implement secure token generation and expiry.
- Ensure robust testing for endpoint security.
- Document the API on swaager under Auth for easy frontend integration.

### PR DESCRIPTION
**Pull request by shadijiha**
Closes #245

### CODE REVIEW

**Review by jonathan-githubofficial — APPROVED** [review]
Lg,gj

**Review by Ahed752 — APPROVED** [review]

File: Backend/Backend/Controllers/Requests/ProfileRequests.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+   public class ChangePasswordRequest
+   {
+      [Required]
+      public string OldPassword { get; set; }
+      
+      [Required]
+      public string NewPassword { get; set; }
+      
+      [Required]
+      public string ConfirmNewPassword { get; set; }
```
Comment (BigSaladMan) [inline Backend/Backend/Controllers/Requests/ProfileRequests.cs:16]: Shouldnt this be handled in the frontend?

File: Backend/Backend/Services/ProfileService.cs [hunk 2]
```diff
@@ -180,4 +180,18 @@ public async Task<string> GetUserAvatarNameAsync(Guid userId)
...
         return user.Avatar;
     }
+
+    public async Task ChangePassword(User user, ChangePasswordRequest request) {
+        // Check if old password is ok
+        if (!BCrypt.Net.BCrypt.Verify(request.OldPassword, user!.Password)) {
+            throw new Exception("Invalid Credentials");
+        }
+
+        if (request.NewPassword != request.ConfirmNewPassword)
```
Comment (BigSaladMan) [inline Backend/Backend/Services/ProfileService.cs:190]: This should be handled in the front end logic mostly

**Review by BigSaladMan — APPROVED** [review]
Just some small comments, rest is good

File: Backend/Backend/Controllers/Requests/ProfileRequests.cs (hunk 1 above)
Comment (mathieutrudeau) [reply Backend/Backend/Controllers/Requests/ProfileRequests.cs:16]: Nothing wrong with redundancy

**Pull request merged**